### PR TITLE
Use explicit return type in data_snippets.cc

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #2604.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2618)

### CODE REVIEW

File: google/cloud/bigtable/examples/data_snippets.cc [hunk 1]
```diff
@@ -163,11 +163,17 @@ void BulkApply(google::cloud::bigtable::Table table, int argc, char* argv[]) {
...
-      throw std::runtime_error(status.message());
+    std::vector<cbt::FailedMutation> failures =
+        table.BulkApply(std::move(bulk));
+    if (failures.empty()) {
+      std::cout << "All mutations applied successfully\n";
+      return;
     }
+    std::cout << "The following mutations failed: ";
+    for (auto const& f : failures) {
+      std::cout << "index[" << f.original_index() << "]=" << f.status() << "\n";
```
Comment (devbww) [inline google/cloud/bigtable/examples/data_snippets.cc]: s/cout/cerr/?

File: google/cloud/bigtable/examples/data_snippets.cc [hunk 2]
```diff
@@ -973,6 +969,37 @@ void InsertTestData(google::cloud::bigtable::Table table, int argc, char*[]) {
   }
 }
 
+// This command just generates data suitable for other examples to run. This
+// code is not extracted into the documentation.
+void PopulateTableHierarchy(google::cloud::bigtable::Table table, int argc,
+                            char* argv[]) {
+  if (argc != 1) {
+    throw Usage{
+        "populate-table-hierarchy: <project-id> <instance-id> <table-id>"};
```
Comment (devbww) [inline google/cloud/bigtable/examples/data_snippets.cc]: s/://

File: google/cloud/bigtable/examples/data_snippets.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/bigtable/examples/data_snippets.cc]: Fixed.

File: google/cloud/bigtable/examples/data_snippets.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/bigtable/examples/data_snippets.cc]: Fixed.

**Comment by coryan** [comment]
Thanks! PTAL.

**Pull request merged**